File: Project Euler/72.cs
using System;

class dkafldjaf
{
	static void Main()
	{
		long z = 0;
		int cnt = 2;
		int[] a= new int[(int)50];
		a[0]=1;
		a[1]=1;
		for ( int i=0; a[cnt-1]<8 ; i++)
		{
			a[cnt]=a[i]+a[i+1];
			cnt++;
			a[cnt]=a[i+1];
			cnt++;
		}
		Console.WriteLine(cnt);
	}
/*		
		int length = 2;
		long size = (long) length*(length-1)/2;
		//long size = (long) Math.Pow(2,5)-1;
		//long size = 55;
		
		Console.WriteLine(size);		
		
		double[] stupidDaniel = new double[size];
		int index = 0;

		for (int i = 1; i < size; i++)
		{
			stupidDaniel[i-1]=1;
		}

		Console.WriteLine((double)stupidDaniel[0] + "d");
		
		for (int j = 1; j<=length; j++)
		{
			for (int i = 1; i<j; i++)
			{
*//*
				int flag = 0;
				for(int k = 0; k <= index; k++)
				{
					if ((double)i/j == stupidDaniel[k])
					{
						flag = k;
						break;
					}
				}
				
				//Console.WriteLine(flag + " flag " + (double)i/j + " i/j " + index + " index");
				
				if(flag == 0)
				{
					stupidDaniel[index] = (double) i/j;
					index++;
					Console.WriteLine(index + " index");
				
					//Array.Sort(stupidDaniel);
					Console.WriteLine((double)stupidDaniel[0]);
				//}
				
			
			}
		}
//		Console.WriteLine(index);
		
		Array.Sort(stupidDaniel);
		//stupidDaniel = stupidDaniel.Sort;
//		Console.WriteLine((double)stupidDaniel[0]);
//		Console.WriteLine("--");
		for (int k = 1; k<size; k++)
		{
//			Console.WriteLine((double)stupidDaniel[k-1]+ " stupidDaniel");
			if(stupidDaniel[k-1] != stupidDaniel[k])
			{
				z++;
//				Console.WriteLine((double)stupidDaniel[k] + " test");
			}
		}
				Console.WriteLine(z+1);		
	}*/
		/*
		for(int i = 2 ; i<=1e4; i++)
		{
			Console.WriteLine(i);
			for (int j = 1 ; j<i ; j++)
			{
				//Console.WriteLine(i + " " + j);
				if (prime(i))
				{
					z++;
					//Console.WriteLine(z);
				}else if (i%j==0&&j!=1){
				}else if (j==1){
					z++;
					//Console.WriteLine(z);

				}
				else{
					for (int k = 2; k<=	j; k++)
					{
						//Console.WriteLine(k + "--");
						if ( i%k==0&&j%k==0)
						{
							break;
						}
						if ( k==j)
						{
							z++;
							//Console.WriteLine(z);
							break;
						}
					}
				}
			}	
		}
		Console.WriteLine(z);
	}
	
		static bool prime(int x)
	{
		if (x<2)
		{
			return false;
		}
		for(int i = 2; i<= (int)Math.Sqrt(x) ; i++)
		{
			if (x%i==0)
			{
				return false;
			}
		}
		return true;

	}*/
}

File: Project Euler/72.py
#! python
import sys
import math

def relPrime(num1,num2):
	if num2%num1 == 0 and num1 != 1:
		return False
	for i in range(2,math.ceil(math.sqrt(num1))+1):
		if num1%i == 0 and num2%i == 0:
			return False
	return True

end = math.floor(1e6 + 1)
nums = [0] * end
total = 0
for i in range(2,len(nums)):
	k = i-1
	if nums[i] == 0:
		total+=k
		for j in range(2*i,end,i):
			nums[j]+=k
	else:
		k2 = k-nums[i]
		total+=k-nums[i]
		for j in range(2*i,end,i):
			nums[j]+=k2

	#print(i,total)

print(total)
sys.exit(0)
cnt = 0
for i in range(2,21):
	cnt2 = 0
	for j in range(1,i):
		if relPrime(j,i):
			cnt+=1
			cnt2+=1
	print(cnt2)
print(cnt)


sys.exit(0)
array = [1,1]

index = 0
while(True):
	num = array[index] + array[index + 1]
	print(num)
	if index%2 == 1 and num > 1000000:
		break
	array.append(num)
	array.append(array[index + 1])
	index+=1

print("done")